c-dependencies/js-compute-runtime/builtins/text-encoder.cpp

```diff
@@ -0,0 +1,81 @@
+#include "builtins/text-encoder.h"
+
+#include "js/ArrayBuffer.h"
+
+// TODO: remove these once the warnings are fixed
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Winvalid-offsetof"
+#include "js/experimental/TypedData.h"
+#pragma clang diagnostic pop
+
+namespace builtins {
+
+bool TextEncoder::encode(JSContext *cx, unsigned argc, JS::Value *vp) {
+  METHOD_HEADER(1)
+
+  // Default to empty string if no input is given.
+  if (args.get(0).isUndefined()) {
+    JS::RootedObject byte_array(cx, JS_NewUint8Array(cx, 0));
+    if (!byte_array)
+      return false;
```

Comment:
Non-Blocking Question/Comment When would this actually occur? Does this only occur if we are out of memory? If so, we should probably report this as a JS InternalError, which is how this is reported in other runtimes. "InternalError: allocation size overflow"

Reply:
I think that's a totally reasonable change. We should take a pass over the builtins now that they're all split out of js-compute-builtins.cpp and look for similar patterns to fix.